src/javascript/app/base/network_monitor.js

```diff
@@ -30,24 +30,27 @@ const NetworkMonitor = (() => {
         el_tooltip,
         network_status;
 
-    const init = () => {
-        ws_config  = $.extend({ wsEvent, isOnline }, BinarySocketGeneral.initOptions());
-        el_status  = getElementById('network_status');
-        el_tooltip = el_status.parentNode;
-
-        if ('onLine' in navigator) {
-            window.addEventListener('online',  setStatus);
-            window.addEventListener('offline', setStatus);
-        } else { // if not supported, default to online and fallback to WS checks
-            navigator.onLine = true;
-        }
+    const init = () => (
+        new Promise((resolve) => {
+            ws_config  = $.extend({ wsEvent, isOnline }, BinarySocketGeneral.initOptions());
+            el_status  = getElementById('network_status');
+            el_tooltip = el_status.parentNode;
+
+            if ('onLine' in navigator) {
+                window.addEventListener('online',  setStatus);
+                window.addEventListener('offline', setStatus);
+            } else { // if not supported, default to online and fallback to WS checks
+                navigator.onLine = true;
+            }
 
-        if (isOnline()) {
-            BinarySocket.init(ws_config);
-        }
+            if (isOnline()) {
+                BinarySocket.init(ws_config);
+                resolve(); // need to know when socket init is done
```

Comment:
do we need to handle `reject` as well? 🤔

Reply:
you're right, if it's not online it shouldn't wait indefinitely. it should resolve when wsReconnect manages to call binarysocket.init. I can't think of a good way to handle that. let me think about it.

Reply:
I think a better way to handle this would be to move the whole thing to the onOpen event of socket. So we can wait and be sure that socket was definitely initialized and opened before calling wait for authroize. So I'll do that instead.